rotate-preservation-system-password/src/main/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/Lambda.scala

```diff
@@ -35,12 +35,16 @@ class Lambda extends LambdaRunner[RotationEvent, Unit, Config, Dependencies] {
       _ <- IO.raiseWhen(versions(token).contains(Current))(new Exception(s"Secret $secretId is already at AWSCURRENT."))
       _ <- IO.raiseWhen(!versions(token).contains(Pending))(new Exception(s"Secret version $token not set as AWSPENDING for rotation of secret $secretId."))
       _ <- event.step match
-        case CreateSecret => createSecret(client, token, config.apiUser)
-        case SetSecret    => setSecret(client, dependencies.userClient, config.apiUser)
-        case TestSecret   => dependencies.userClient.testNewPassword()
+        case CreateSecret => createSecret(client, token)
+        case SetSecret    => setSecret(client, userClient)
+        case TestSecret   => userClient.testNewPassword()
         case FinishSecret => finishSecret(client, token)
     } yield ()
 
+  private def getAuthDetailsFromSecret(client: DASecretsManagerClient[IO], stage: Stage) = for {
+    currentSecret <- client.getSecretValue[Map[String, String]](stage)
```

Comment:
What do you think about this:

```
  private def getAuthDetailsFromSecret(client: DASecretsManagerClient[IO], stage: Stage) = for {
    (username, password) <- client.getSecretValue[Map[String, String]](stage).map(_.head)
  } yield AuthDetails(username, password)
```

Reply:
That's better, gets rid of `._1` and `._2` thanks.